odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/EnumValueServiceImpl.java

```diff
@@ -1,41 +1,73 @@
 package org.opendatadiscovery.oddplatform.service;
 
 import java.util.List;
+import java.util.Objects;
 import lombok.RequiredArgsConstructor;
-import org.opendatadiscovery.oddplatform.api.contract.model.EnumValue;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.BooleanUtils;
 import org.opendatadiscovery.oddplatform.api.contract.model.EnumValueFormData;
 import org.opendatadiscovery.oddplatform.api.contract.model.EnumValueList;
 import org.opendatadiscovery.oddplatform.mapper.EnumValueMapper;
-import org.opendatadiscovery.oddplatform.repository.EnumValueRepository;
+import org.opendatadiscovery.oddplatform.model.tables.pojos.EnumValuePojo;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveEnumValueRepository;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 @Service
 @RequiredArgsConstructor
 public class EnumValueServiceImpl implements EnumValueService {
-    private final EnumValueRepository enumValueRepository;
+    private final ReactiveEnumValueRepository reactiveEnumValueRepository;
     private final EnumValueMapper mapper;
 
     @Override
     public Mono<EnumValueList> createEnumValues(final Long datasetFieldId, final List<EnumValueFormData> formData) {
-        return Flux.fromStream(formData.stream())
-            .map(fd -> mapper.mapForm(fd, datasetFieldId))
+        final var source =  Flux.fromStream(formData.stream())
+            .map(fd -> mapper.mapToPojo(fd, datasetFieldId)).cache();
+
+        final var hasDuplicates = source
+            .map(EnumValuePojo::getName)
+            .collectList()
+            .map(this::hasDuplicates)
+            .filter(BooleanUtils::isFalse)
+            .switchIfEmpty(Mono.error(new RuntimeException("There are duplicates in enum values")));
```

Comment:
@damirabdul This is probably a convoluted way to simply throw error for duplicates, looking for your suggestions on how to improve this.